src/ast.c

```diff
@@ -731,6 +731,7 @@ jl_value_t *jl_parse_eval_all(const char *fname, size_t len,
         fl_free_gc_handles(fl_ctx, 1);
     }
     else {
+        assert(memchr(fname, 0, len) == NULL); // was checked already in jl_load
```

Comment:
FWIW, these functions should not take a length argument since they really don't support anything other than NUL terminated strings....